react-app/src/library/services/headset.ts

```diff
@@ -100,10 +126,18 @@ export default class HeadsetService {
       this.selectedImplementation.disconnect();
     }
 
+    if (implementation?.vendorName === 'Jabra') {
+      (await this.jabraSdk).deviceList.subscribe((devices) => {
+        if (devices.find((device) => deviceLabel.includes(device.name.toLowerCase()))) {
+          this.emit('jabraPermissionRequested', { webHidPairing: webHidPairing });
```

Comment:
This block here is checking to see if the selected device was already granted permissions to connect to WebHID as that selection will persist on a site.  So we subscribe to `deviceList` which is all the Jabra devices/WebHID enabled devices that were already granted permissions and cycle through to determine if the selected device was already granted permissions.  If it wasn't yet, then we would emit the event that the SDK would listen to and react accordingly.  Honestly I am little iffy with using the `.subscribe` with `deviceList` here... `deviceList` also offers a `forEach` but if I remember correctly that one was a little finicky. `subscribe` was far more consistent.

Reply:
Gotcha. Since this is a subscribe, make sure you unsubscribe when you are done. Right now, we will subscribe every time `changeImplementation` is called. So if it happens to get called three times, we will emit `this.emit('jabraPermissionRequested', { webHidPairing: webHidPairing });` three times.

Reply:
Hmm. That's a good point.  Let me try and get the `forEach` to maybe work, that should hopefully take care of that issue